contrib/mongodb/mongo-go-driver/mongo/mongo.go

```diff
@@ -0,0 +1,104 @@
+// Package mongo provides functions to trace the mongodb/mongo-go-driver package (https://github.com/mongodb/mongo-go-driver).
+//
+// `NewMonitor` will return an event.CommandMonitor which is used to trace
+// requests.
+package mongo
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/mongodb/mongo-go-driver/core/event"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// The values of these mongo fields will not be scrubbed out. This allows the
+// non-sensitive collection names to be captured.
+var unscrubbedFields = []string{
+	"ordered", "insert", "count", "find", "create",
+}
+
+type spanKey struct {
+	ConnectionID string
+	RequestID    int64
+}
+
+type monitor struct {
+	sync.Mutex
+	spans map[spanKey]ddtrace.Span
+}
+
+func (m *monitor) Started(evt *event.CommandStartedEvent) {
+	hostname, port := peerInfo(evt)
+	statement := scrub(evt.Command).ToExtJSON(false)
+
+	span, _ := tracer.StartSpanFromContext(evt.Context, "mongodb.query",
+		tracer.ServiceName("mongo"),
+		tracer.ResourceName("mongo."+evt.CommandName),
+		tracer.Tag(ext.DBInstance, evt.DatabaseName),
+		tracer.Tag(ext.DBStatement, statement),
```

Comment:
Although the tag name you've chosen is the correct standard used in most tracers, we have to use `mongodb.query`. Our choice of tag naming was rather uneven, you can see the span types and tag names we use in the agent [here](https://github.com/gbbr/datadog-trace-agent/blob/master/obfuscate/obfuscate.go#L37).

That said, it might be best to remove this constant. It might encourage others to use it too.